projects/bp-gallery/src/components/views/picture/sidebar/comments/FormattedComment.tsx

```diff
@@ -43,19 +59,97 @@ const FormattedComment = ({ comment }: { comment: FlatComment }) => {
     onCompleted: () => setPinned(false),
   });
 
+  const [deleteComment] = useDeclineCommentMutation({
+    refetchQueries: ['getPictureInfo'],
+  });
+
+  const onDelete = useCallback(async () => {
+    const getDescendants = (comment: FlatComment, descendants: string[] = []): string[] => {
+      if (!comment.childComments) return [];
+      descendants.push(...comment.childComments.map(childComment => childComment.id));
+      comment.childComments.forEach(childComment => getDescendants(childComment, descendants));
+      return descendants;
+    };
```

Comment:
This could probably be moved outside the component as it doesn't depend on any local variables.

Reply:
Done